training/train.py

```diff
@@ -354,59 +320,123 @@ def get_deployed_model_version(yaml_data, deployment_prefix):
-                  'Criteria for promotion was [current_recall_at_30 >= prev_recall_at_30].'.format(
-                      previous_version,
-                      json.dumps(prev_hyperparams, indent=4, sort_keys=True),
-                      MODEL_VERSION,
-                      json.dumps(hyper_params, indent=4, sort_keys=True))
-    description = description.replace('"', '\\"').replace('\n', '\\n')
+    upstream_repo = Github(GITHUB_TOKEN).get_repo(f'{UPSTREAM_REPO_NAME}/{PROJECT_NAME}')
+    deployed_version, deployed_hyperparams, yaml_dict = read_deployed_data(upstream_repo,
+                                                                           s3_client)
 
     recall_at_30 = hyper_params['recall_at_30']
-    prev_recall = prev_hyperparams.get('recall_at_30', 0.55)
-    _logger.info('create_git_pr:: Prev => Model {}, Recall {}  Curr => Model {}, Recall {}'.format(
-        previous_version, prev_recall, MODEL_VERSION, recall_at_30))
-    if recall_at_30 >= prev_recall:
-        try:
-            # 1. Get the bash script from rudra to raise PR
-            exec_command(["wget", "-v",
-                          "https://raw.githubusercontent.com/fabric8-analytics/"
-                          "fabric8-analytics-rudra/master/rudra/utils/github_helper.sh",
-                          "-O", "./github_helper.sh"], 60)
-
-            # 2. Provide execute permission to the github script.
-            exec_command(["chmod", "+x", "./github_helper.sh"], 30)
-
-            # 3. Invoke bash script to create a saas-analytics PR
-            exec_command(["./github_helper.sh", "f8a-pypi-insights.yaml", "MODEL_VERSION",
-                          str(MODEL_VERSION), description], 60)
-        except Exception as e:
-            _logger.error('ERROR - execute command raise exception')
-            raise e
+    deployed_recall_at_30 = deployed_hyperparams.get('recall_at_30', 0.55)
+    _logger.info('create_git_pr:: Deployed => Model %s, Recall %f Current => Model %s, Recall %f',
+                 deployed_version, deployed_recall_at_30, MODEL_VERSION, recall_at_30)
+    if recall_at_30 >= deployed_recall_at_30:
+        promotion_creteria = 'current_recall_at_30 >= deployed_recall_at_30'
+
+        params = hyper_params.copy()
+        params.update({'promotion_criteria': str(promotion_creteria)})
+        branch_name, commit_message = create_branch_and_update_yaml(yaml_dict, params,
+                                                                    deployed_version)
+
+        hyper_params_formated = build_hyper_params_message(hyper_params)
+        prev_hyper_params_formated = build_hyper_params_message(deployed_hyperparams)
+        body = f'Current deployed model details:\n- Model version :: `{deployed_version}`\n' \
+               f'{prev_hyper_params_formated}New model details:\n' \
+               f'- Model version :: `{MODEL_VERSION}`\n' \
+               f'{hyper_params_formated}Criteria for promotion is `{promotion_creteria}`\n'
```

Comment:
Why not multi line string?
f"""Current deployed model details:
Model version: {deployed_version}
...
"""